src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx

```diff
@@ -0,0 +1,214 @@
+import { generatePath, useParams } from 'react-router-dom';
+import camelCase from 'lodash/camelCase';
+
+export default function ReleaseApiDataSetLocationsMappingPage() {
+  const { dataSetId, releaseId, publicationId } =
+    useParams<ReleaseDataSetRouteParams>();
+
+  const { data: dataSet, isLoading: isLoadingDataSet } = useQuery(
+    apiDataSetQueries.get(dataSetId),
+  );
+
+  const {
+    data: locationsMapping = { levels: {} },
+    isLoading: isLoadingMapping,
+  } = useQuery({
+    ...apiDataSetVersionQueries.getLocationsMapping(
+      dataSet?.draftVersion?.id ?? '',
+    ),
+    enabled: !!dataSet?.draftVersion?.id,
+  });
+
+  const {
+    autoMappedLocations,
+    newLocationCandidates,
+    unmappedAndManuallyMappedLocations,
+  } = getApiDataSetLocationMappings(locationsMapping);
+
+  const totalNewLocationCandidates = Object.values(
+    newLocationCandidates,
+  ).flatMap(key => key).length;
+
+  const unmappedLocationErrors = getUnmappedLocationErrors(
+    unmappedAndManuallyMappedLocations,
+  );
+
+  const navItems: NavItem[] = [
+    {
+      id: 'unmapped-locations',
+      text: 'Locations not found in the new data set',
+      subNavItems: Object.keys(unmappedAndManuallyMappedLocations).map(
+        level => {
+          return {
+            id: `unmapped-${level}`,
+            text: locationLevelsMap[camelCase(level)]?.plural ?? level,
+          };
+        },
+      ),
+    },
+    {
+      id: 'new-locations',
+      text: 'New locations',
+      subNavItems: Object.keys(newLocationCandidates).map(level => {
+        return {
+          id: `new-${level}`,
+          text: locationLevelsMap[camelCase(level)]?.plural ?? level,
+        };
+      }),
+    },
+    { id: 'auto-mapped-locations', text: 'Auto mapped locations' },
+  ];
+
+  return (
+    <>
+      <Link
+        back
+        className="govuk-!-margin-bottom-6"
+        to={generatePath<ReleaseDataSetRouteParams>(
+          releaseApiDataSetDetailsRoute.path,
+          {
+            publicationId,
+            releaseId,
+            dataSetId,
+          },
+        )}
+      >
+        Back
+      </Link>
+      <LoadingSpinner loading={isLoadingDataSet || isLoadingMapping}>
+        <div className="govuk-grid-row">
+          <PageNav items={navItems} />
+          <div className="govuk-grid-column-three-quarters">
+            <span className="govuk-caption-l">Map locations</span>
+            <h2>{dataSet?.title}</h2>
+            {!!unmappedLocationErrors.length && (
+              <ErrorSummary
+                errors={unmappedLocationErrors}
+                headingTag="h3"
+                title="Unmapped locations"
+                updateDocumentTitle={false}
+              />
+            )}
+            <p>
+              Existing locations in the publication data set should be mapped to
+              an equivalent location in the new data set. We try to
+              automatically match up locations that have the same name, but you
```

Comment:
`name and codes` strictly speaking, if you feel like that needs feeding back in?